### DISCUSSION TITLE
Branching strategy

### DISCUSSION THREAD

**Discussion by dinophile** [cds-snc/url-shortener#31] [Ideas]
Would love it if we can document and agree on things like branch naming, PR expectations, I assume we'll run with the trunk development/PR flow? @jeberhardt @sylviamclaughlin @patheard @ascheid @feralbob

**Comment by patheard** [comment]
Yeah trunk based dev with a flow something like this would be my preference:

1. PR to `main` is a planned deployment to **Staging**.
2. Merge to `main` deploys app and infra changes to **Staging**.
3. PR to `main` with changes to [version manifest files](https://github.com/cds-snc/scan-files/tree/main/.github/manifests) is a planned deployment to **Prod**.  These manifests would contain Docker image tags and repo release tags.
4. Merge to `main` of these version manifest files deploys app and infra changes to **Prod**.

I like this flow because it makes deployments to Staging easy and gives good control/visibility over what is deployed to **Prod**.  

We could also set code owners on the version manifests files so specific people needed approval before a deploy to **Prod** could happen.

**Reply by jeberhardt** [reply]
Yeah, this is the approach taken by **GC-Articles**, and it worked well. A few _npm_ scripts as a part of the project handled updating version files, creating the required PRs, and triggering Github actions.

**Comment by dinophile** [comment]
Everyone in favour? Say Aye...or just add a 👍 that works too!

@feralbob and @sylviamclaughlin
